operator/v1alpha1/types_kueue.go

```diff
@@ -0,0 +1,305 @@
+)
+
+// +kubebuilder:validation:Enum="Enabled;Disabled"
+type EnabledOrDisabled string
+
+const (
+	Enabled  EnabledOrDisabled = "Enabled"
+	Disabled EnabledOrDisabled = "Disabled"
+)
+
+type KueueConfiguration struct {
+	// waitForPodsReady configures gang admission
+	// +optional
+	WaitForPodsReady WaitForPodsReady `json:"waitForPodsReady,omitempty"`
+	// integrations are the types of integrations Kueue will manager
+	// +required
+	Integrations Integrations `json:"integrations"`
+	// featureGates are advanced features for Kueue
+	// +optional
+	FeatureGates map[string]EnabledOrDisabled `json:"featureGates,omitempty"`
+	// resources provides additional configuration options for handling the resources.
+	// Supports https://github.com/kubernetes-sigs/kueue/blob/release-0.10/keps/2937-resource-transformer/README.md
+	// +optional
+	Resources Resources `json:"resources,omitempty"`
+	// manageJobsWithoutQueueName controls whether or not Kueue reconciles
+	// jobs that don't set the annotation kueue.x-k8s.io/queue-name.
+	// Allowed values are NoQueueName and QueueName
+	// Default will be QueueName
+	// +kubebuilder:default=QueueName
+	// +optional
+	ManageJobsWithoutQueueName *ManageJobsWithoutQueueNameOption `json:"manageJobsWithoutQueueName,omitempty"`
+	// managedJobsNamespaceSelector can be used to omit some namespaces from ManagedJobsWithoutQueueName
+	// Only valid if ManagedJobsWithoutQueueName is NoQueueName
+	// +optional
+	ManagedJobsNamespaceSelector *metav1.LabelSelector `json:"managedJobsNamespaceSelector,omitempty"`
+	// fairSharing controls the fair sharing semantics across the cluster.
+	// +optional
+	FairSharing FairSharing `json:"fairSharing,omitempty"`
+	// metrics allows one to change if metrics
+	// are enabled or disabled.
+	// Microshift does not enable metrics by default
+	// Default will assume metrics are enabled.
+	// +kubebuilder:default=Enabled
+	// +optional
+	Metrics *EnabledOrDisabled `json:"metrics,omitempty"`
+}
+
+// KueueStatus defines the observed state of Kueue
+type KueueStatus struct {
+	operatorv1.OperatorStatus `json:",inline"`
+}
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// KueueList contains a list of Kueue
+//
+// Compatibility level 4: No compatibility is provided, the API can change at any point for any reason. These capabilities should not be used by applications needing long term support.
+// +openshift:compatibility-gen:level=4
+type KueueList struct {
+	metav1.TypeMeta `json:",inline"`
+	// metadata for the list
+	// +optional
+	metav1.ListMeta `json:"metadata,omitempty"`
+	// items is a slice of kueue
+	// this is a cluster scoped resource and there can only be 1 kueue
+	// +kubebuilder:validation:MaxItems=1
+	// +required
+	Items []Kueue `json:"items"`
+}
+
+// These structs come directly from Kueue.
+type Resources struct {
+	// excludeResourcePrefixes defines which resources should be ignored by Kueue
```

Comment:
@JoelSpeed I don't have a clue what to put here for MaxLength.

These are left vague as they are overrides to abstract away GPUs to accelerator memory (is on example).

Is there a way to turn off the linter for a few of these?

Reply:
Kueue doesn't specify a maximum length for these, but if you needed one for OpenShift I think 16 is more than enough.   The motivating example was NVIDIA MIG; current GPUs 8 entries is more than enough.